tests/test_merge_dynamic_streams_block.py

```diff
@@ -0,0 +1,163 @@
+            self.last_notified[DEFAULT_TERMINAL][3].to_dict(),
+            {"D": "d", "B": "b"})
+        self.assertDictContainsSubset(
+            self.last_notified[DEFAULT_TERMINAL][4].to_dict(),
+            {"D": "d", "E": "e"})
+
+    def test_with_expiration(self):
+        blk = MergeDynamicStreams()
+        self.configure_block(blk, {
+            "stream": "{{ $stream }}",
+            "expiration": {"seconds": 0.1},
+        })
+        blk.start()
+        self.process_test_signals(blk)
+        blk.stop()
+        self.assert_num_signals_notified(5)
+        self.assertDictEqual(
+            self.last_notified[DEFAULT_TERMINAL][0].to_dict(),
+            {"A": "a", "stream": 1})
+        self.assertDictContainsSubset(
+            self.last_notified[DEFAULT_TERMINAL][1].to_dict(),
+            {"B": "b", "stream": 2})
+        self.assertDictContainsSubset(
+            self.last_notified[DEFAULT_TERMINAL][2].to_dict(),
+            {"C": "c", "B": "b"})
+        self.assertDictContainsSubset(
+            self.last_notified[DEFAULT_TERMINAL][3].to_dict(),
+            {"D": "d", "B": "b"})
+        self.assertDictContainsSubset(
+            self.last_notified[DEFAULT_TERMINAL][4].to_dict(),
+            {"D": "d", "E": "e"})
+
+    def test_signal_expiration_job(self):
+        blk = MergeDynamicStreams()
+        blk._signals["group"]["stream"] = Signal()
+        blk._expiration_jobs["group"]["stream"] = MagicMock()
+        self.assertTrue("stream" in blk._signals["group"])
+        self.assertTrue("stream" in blk._expiration_jobs["group"])
+        blk._signal_expiration_job("group", "stream")
+        self.assertFalse("stream" in blk._signals["group"])
+        self.assertFalse("stream" in blk._expiration_jobs["group"])
+
+    def test_reset_expiration_job_on_new_signal_input_1(self):
+        """ Signal expiration job is not called if new signals come in """
+        blk = MergeDynamicStreams()
+        blk._signal_expiration_job = MagicMock()
+        self.configure_block(blk, {
+            "stream": "{{ $stream }}",
+            "expiration": {"seconds": 0.1},
+        })
+        blk.start()
+        blk.process_signals([Signal({"A": "a", "stream": 1})])
+        sleep(0.05)
+        blk.process_signals([Signal({"B": "b", "stream": 1})])
+        sleep(0.05)
+        blk.process_signals([Signal({"C": "c", "stream": 1})])
+        sleep(0.05)
+        blk.stop()
+        self.assertEqual(blk._signal_expiration_job.call_count, 0)
+
+    def test_reset_expiration_job_on_new_signal_input_2(self):
+        """ Signal expiration job is not called if new signals come in """
+        blk = MergeDynamicStreams()
+        blk._signal_expiration_job = MagicMock()
+        self.configure_block(blk, {
+            "stream": "{{ $stream }}",
+            "expiration": {"seconds": 0.1},
+        })
+        blk.start()
+        blk.process_signals([Signal({"A": "a", "stream": 2})])
+        sleep(0.05)
```

Comment:
Similar sleep comment here. Can we test this without sleeping and/or relying on the scheduler to be accurate?